main.go

```diff
@@ -1,12 +1,95 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"encoding/base64"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
 
+	"github.com/ollama/ollama/auth"
 	"github.com/ollama/ollama/cmd"
-	"github.com/spf13/cobra"
+	"github.com/pkg/browser"
 )
 
 func main() {
-	cobra.CheckErr(cmd.NewCLI().ExecuteContext(context.Background()))
+	if err := cmd.NewCLI().ExecuteContext(context.Background()); err != nil {
+		fmt.Fprintln(os.Stderr, "Error:", err)
+
+		if strings.Contains(err.Error(), "unknown ollama key") {
+			if err := handleUnkownKeyError(err); err != nil {
+				fmt.Fprintln(os.Stderr, "Error:", err)
+			}
+		}
+
+		os.Exit(1)
+	}
+}
+
+func handleUnkownKeyError(accessErr error) error {
+	// find SSH public key in the error message
+	sshKeyPattern := `ssh-\w+ [^\s"]+`
+	re := regexp.MustCompile(sshKeyPattern)
+	matches := re.FindStringSubmatch(accessErr.Error())
```

Comment:
who owns the root cause line of accessErr? If we do or know who does, can we find a better way than regexp matches on the message?

Reply:
Ollama owns this, but this is across the cli/ollama boundary as well, so all we have is the error string.